### PR TITLE
Minor changes release production

### PR DESCRIPTION
**Pull request by mrossi113**
## Description
Cleaning up the deployment documentation for developer and release steps.

## Related Issue / Ticket

- [JIRA issue]()
- [Github issue]()

## Steps to Test

1. Review the changes to make sure it make sense and accurate. 

## Screenshots
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.

## Additional Notes:

Anything else to add?

#### Impacted Areas in Application

* 

#### @TODO

*

### CODE REVIEW

**Comment by mrossi113** [comment]
Should we remove the following script `scripts/deploy-gh-pages.sh` because we no longer can do this?

File: README.md [hunk 1]
```diff
@@ -18,8 +18,8 @@ Please follow the steps in [Contributing docs](.github/CONTRIBUTING.md) to set u
 
 ### Developer Deployment
 
-Please see [Deployment docs](docs/developer-deploy.md) for steps on deploying development work to a fork of Mayflower's Github Pages.
- 
+Every time a branch passes CircleCI it will deploy the code to a staging environment by creating the following URL `http://mayflower.digital.mass.gov/b/<branch name>/index.html`. Which will allow the developer to share the URL with reviewers to test their code.
```
Comment (yriahi) [inline README.md]: _Re: ...to a staging environment..._ 
We don't have a staging environment; but we can re-phrase to indicate that code gets pushed to the "/b" folder where all branches are deployed.

**Comment by yriahi** [comment]
Github pages deployments is deprecated and is no longer supported. Let's remove that!

**Review by yriahi — APPROVED** [review]
looks good to me 👍

**Pull request merged**